### PR TITLE
Allow using woocommerce icons for product details section description

### PR DESCRIPTION
**Pull request by nathanss**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This allows us to add woocommerce icons (added in the icons folder of product-editor) to product templates. We currently have designs that use not only WordPress icons, but some additional icons, so this will be useful.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Install and activate the following plugin:
[wooicons-test.zip](https://github.com/woocommerce/woocommerce/files/14602181/wooicons-test.zip)
2. Enable the new product editor at /wp-admin/admin.php?page=wc-settings&tab=advanced&section=features
3. Go to Products > Add
4. Click on "Change product type" 
5. Verify that the "gift" icon looks correct in the popover

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/icons/gift.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import { createElement } from '@wordpress/element';
+
+const gift = (
+	<svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (retrofox) [inline packages/js/product-editor/src/icons/gift.tsx]: we should use core components here. You can take a look at the [ShoppingBags](https://github.com/woocommerce/woocommerce/blob/7fe00de7278cbc0d7806e1733b8df7320feef933/packages/js/product-editor/src/images/shopping-bags/index.tsx) image.

**Review by retrofox — COMMENTED** [review]
I left a comment about using core components to define the Icon. Also, adding a story wont hurt ;-). Anyway it isn't required of course.

**Comment by nathanss** [comment]
> I left a comment about using core components to define the Icon.

Thanks! I implemented it.

> Also, adding a story wont hurt ;-). Anyway it isn't required of course.

I've thought about it, but I thought it would be a significant effort. I think we can revisit once we have some more icons on that folder (there's currently only 3), and add something like this: https://wordpress.github.io/gutenberg/?path=/docs/icons-icon--docs

File: packages/js/product-editor/src/icons/gift.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const gift = (
+	<SVG viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+		<Rect
+			x="-0.75"
+			y="-0.75"
+			fill="none"
+			width="9.5"
+			height="14.5"
+			transform="matrix(3.97376e-08 -1 -1 -4.80825e-08 18.5 18.5)"
+			stroke="#1E1E1E"
```
Comment (retrofox) [inline packages/js/product-editor/src/icons/gift.tsx:16]: Hardcoding the color doesn't feel great. It seems that all icons do that, right?

**Review by retrofox — APPROVED** [review]
LGTM :shipit:

**Comment by retrofox** [comment]
> > I left a comment about using core components to define the Icon.
> 
> Thanks! I implemented it.
> 
> > Also, adding a story wont hurt ;-). Anyway it isn't required of course.
> 
> I've thought about it, but I thought it would be a significant effort. I think we can revisit once we have some more icons on that folder (there's currently only 3), and add something like this: https://wordpress.github.io/gutenberg/?path=/docs/icons-icon--docs

Maybe we can create an issue and address it in a follow-up.

**Comment by nathanss** [comment]
@retrofox I will create a follow up issue tomorrow!

**Pull request merged**